pkg/ccl/backupccl/restore_job.go

```diff
@@ -247,6 +247,7 @@ func makeBackupLocalityMap(
 					return nil, errors.Wrap(err,
 						"creating locality external storage configuration")
 				}
+				conf.URI = uri
```

Comment:
As a follow-up, do we want to put some validation somewhere either in LinkExternalSSTable or in the online restore code that we are dealing with a valid external file?